castai/resource_allocation_group.go

```diff
@@ -0,0 +1,303 @@
+package castai
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/hashicorp/go-uuid"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+
+	"github.com/castai/terraform-provider-castai/castai/sdk"
+)
+
+func resourceAllocationGroup() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceAllocationGroupCreate,
+		ReadContext:   resourceAllocationGroupRead,
+		UpdateContext: resourceAllocationGroupUpdate,
+		DeleteContext: resourceAllocationGroupDelete,
+		CustomizeDiff: resourceAllocationGroupDiff,
+		Description:   "Manage allocation group. Allocation group [reference](https://docs.cast.ai/docs/allocation-groups)",
+		Importer: &schema.ResourceImporter{
+			StateContext: allocationGroupImporter,
```

Comment:
You seem to just use ID to get the resource. Why custom importer? I think it would work with this
```suggestion
			StateContext: schema.ImportStatePassthroughContext,
```